implementation/src/main/java/io/smallrye/metrics/MetricsRegistryImpl.java

```diff
@@ -556,7 +556,7 @@ private boolean isSameType(Metric metricInstance, MetricType type) {
     }
 
     @Override
-    public Map<String, Metadata> getMetadata() {
+    public synchronized Map<String, Metadata> getMetadata() {
```

Comment:
Is the copy necessary?
I expect few updates and many reads on this map, could it be better represented by a `ConcurrentHashMap`?

Reply:
You mean we would return an unmodifiable view here? Hmmm, that's a good point, but on the other hand, all other access to this map is in synchronized methods (`get`, `register`... - and they will have to stay synchronized because they operate on multiple maps at the same time), so I suppose making it a `ConcurrentHashMap` could unnecessarily slow those methods down, and they will probably be called more often than `getMetadata`. So it's hard to say what's better.